src/tests/BuildWasmApps/Wasm.Build.Tests/WasmRunOutOfAppBundleTests.cs

```diff
@@ -0,0 +1,60 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.IO;
+using Xunit;
+using Xunit.Abstractions;
+
+#nullable enable
+
+namespace Wasm.Build.Tests
+{
+    public class WasmRunOutOfAppBundleTests : BuildTestBase
+    {
+        public WasmRunOutOfAppBundleTests(ITestOutputHelper output, SharedBuildPerTestClassFixture buildContext) : base(output, buildContext)
+        {}
+
+        [Theory]
+        [BuildAndRun(aot: false, host: RunHost.V8)]
+        [BuildAndRun(aot: true, host: RunHost.V8)]
+        [BuildAndRun(aot: false, host: RunHost.Chrome)]
+        [BuildAndRun(aot: true, host: RunHost.Chrome)]
+        [BuildAndRun(aot: false, host: RunHost.NodeJS)]
+        [BuildAndRun(aot: true, host: RunHost.NodeJS)]
```

Comment:
You can use `[BuildAndRun]`, and it will add all these combinations by default.

Reply:
If we replace it with `[BuildAndRun]`, we will get an error like this.
`InitProjectDir` will fail because the test is performed on the same id.

https://github.com/dotnet/runtime/blob/2db797485e4a885aae7572806090c42accc2c62f/src/tests/BuildWasmApps/Wasm.Build.Tests/BuildTestBase.cs#L281

```
  ==============================================================================================
  =============== Running with Workloads ===============
  ==============================================================================================

      Wasm.Build.Tests.WasmRunOutOfAppBundleTests.RunOutOfAppBundle(buildArgs: BuildArgs { ProjectName = placeholder, Config = Debug, AOT = False, ProjectFileContents = placeholder, ExtraBuildArgs =  }, host: V8, id: "2kfi50gc.hjx") [STARTING]
      Wasm.Build.Tests.ConfigSrcTests.ConfigSrcAbsolutePath(buildArgs: BuildArgs { ProjectName = placeholder, Config = Debug, AOT = False, ProjectFileContents = placeholder, ExtraBuildArgs =  }, host: V8, id: "jnopof1n.ffi") [STARTING]
      Wasm.Build.Tests.WasmRunOutOfAppBundleTests.RunOutOfAppBundle(buildArgs: BuildArgs { ProjectName = placeholder, Config = Debug, AOT = False, ProjectFileContents = placeholder, ExtraBuildArgs =  }, host: V8, id: "2kfi50gc.hjx") [FINISHED] Time: 8.4030997s
      Wasm.Build.Tests.WasmRunOutOfAppBundleTests.RunOutOfAppBundle(buildArgs: BuildArgs { ProjectName = placeholder, Config = Debug, AOT = False, ProjectFileContents = placeholder, ExtraBuildArgs =  }, host: Chrome, id: "2kfi50gc.hjx") [STARTING]
      Wasm.Build.Tests.ConfigSrcTests.ConfigSrcAbsolutePath(buildArgs: BuildArgs { ProjectName = placeholder, Config = Debug, AOT = False, ProjectFileContents = placeholder, ExtraBuildArgs =  }, host: V8, id: "jnopof1n.ffi") [FINISHED] Time: 8.4034685s
      Wasm.Build.Tests.ConfigSrcTests.ConfigSrcAbsolutePath(buildArgs: BuildArgs { ProjectName = placeholder, Config = Debug, AOT = False, ProjectFileContents = placeholder, ExtraBuildArgs =  }, host: NodeJS, id: "jnopof1n.ffi") [STARTING]
      Wasm.Build.Tests.WasmRunOutOfAppBundleTests.RunOutOfAppBundle(buildArgs: BuildArgs { ProjectName = placeholder, Config = Debug, AOT = False, ProjectFileContents = placeholder, ExtraBuildArgs =  }, host: Chrome, id: "2kfi50gc.hjx") [FAIL]
        System.IO.IOException : The file '/Users/yk-yamada/Projects/github.com/yamachu/dotnet-runtime/artifacts/bin/Wasm.Build.Tests/Release/net7.0/browser-wasm/2kfi50gc.hjx/nuget.config' already exists.
        Stack Trace:
             at Interop.ThrowExceptionForIoErrno(ErrorInfo errorInfo, String path, Boolean isDirectory)
             at Microsoft.Win32.SafeHandles.SafeFileHandle.Open(String path, OpenFlags flags, Int32 mode, Func`4 createOpenException)
             at Microsoft.Win32.SafeHandles.SafeFileHandle.Open(String fullPath, FileMode mode, FileAccess access, FileShare share, FileOptions options, Int64 preallocationSize, UnixFileMode openPermissions, Int64& fileLength, UnixFileMode& filePermissions, Func`4 createOpenException)
             at System.IO.FileSystem.CopyFile(String sourceFullPath, String destFullPath, Boolean overwrite)
          /Users/yk-yamada/Projects/github.com/yamachu/dotnet-runtime/src/tests/BuildWasmApps/Wasm.Build.Tests/BuildTestBase.cs(289,0): at Wasm.Build.Tests.BuildTestBase.InitProjectDir(String dir)
          /Users/yk-yamada/Projects/github.com/yamachu/dotnet-runtime/src/tests/BuildWasmApps/Wasm.Build.Tests/BuildTestBase.cs(343,0): at Wasm.Build.Tests.BuildTestBase.BuildProject(BuildArgs buildArgs, String id, BuildProjectOptions options)
          /Users/yk-yamada/Projects/github.com/yamachu/dotnet-runtime/src/tests/BuildWasmApps/Wasm.Build.Tests/WasmRunOutOfAppBundleTests.cs(24,0): at Wasm.Build.Tests.WasmRunOutOfAppBundleTests.RunOutOfAppBundle(BuildArgs buildArgs, RunHost host, String id)
             at InvokeStub_WasmRunOutOfAppBundleTests.RunOutOfAppBundle(Object, Object, IntPtr*)
             at System.Reflection.MethodInvoker.Invoke(Object obj, IntPtr* args, BindingFlags invokeAttr)
      Wasm.Build.Tests.WasmRunOutOfAppBundleTests.RunOutOfAppBundle(buildArgs: BuildArgs { ProjectName = placeholder, Config = Debug, AOT = False, ProjectFileContents = placeholder, ExtraBuildArgs =  }, host: Chrome, id: "2kfi50gc.hjx") [FINISHED] Time: 0.0153179s
```